delegate: order AZs in SHOW REGIONS commands

otan opened the pull request:
Release note (sql change): Availability zones were not ordered when
using the SHOW REGIONS set of commands. This has now been changed to be
ordered.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62490)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
Nit: Can we add a test case to validate?

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62490#-:-MWVxqLm9Q0Av9v8dqo-:b-44paes)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
i don't know how to get the zones out of order in logic tests

ajstorm commented:
I can hit it with --global.  Maybe that's a clue?

```

> cockroach demo --nodes 9 --global --empty


            [2021-03-23-6:44:46 pm]

#

# Welcome to the CockroachDB demo database!

#

# You are connected to a temporary, in-memory CockroachDB cluster of 9
nodes.

#

# This demo session will attempt to enable enterprise features

# by acquiring a temporary license from Cockroach Labs in the background.

# To disable this behavior, set the environment variable

# COCKROACH_SKIP_ENABLING_DIAGNOSTIC_REPORTING=true.

#

# Reminder: your changes to data stored in the demo session will not be
saved!

#

# Connection parameters:

#   (webui)
http://127.0.0.1:8080/demologin?password=demo79017&username=demo

#   (sql)      ***@***.***:26257?sslmode=require

#   (sql/unix) postgres://demo:demo79017@
?host=%2Fvar%2Ffolders%2Fk6%2F4p309l956119220lt0m95qw40000gp%2FT%2Fdemo395250020&port=26257

#

# To display connection parameters for other nodes, use \demo ls.

#

# The user "demo" with password "demo79017" has been created. Use it to
access the Web UI!

#

# Server version: CockroachDB CCL v21.1.0-alpha.3-1613-g33711c19e7
(x86_64-apple-darwin19.6.0, built 2021/03/23 21:16:05, go1.15.5) (same
version as client)

# Cluster ID: 8f5025e6-b892-41a4-ae7c-250985a7d3fc

#

# Enter \? for a brief introduction.

#

***@***.***:26257/defaultdb> show regions;

     region    |  zones  | database_names | primary_region_of

---------------+---------+----------------+--------------------

  europe-west1 | {b,d,c} | {}             | {}

  us-east1     | {b,c,d} | {}             | {}

  us-west1     | {b,a,c} | {}             | {}

(3 rows)


Time: 9ms total (execution 9ms / network 0ms)

```

On Tue, Mar 23, 2021 at 5:30 PM Oliver Tan ***@***.***> wrote:

> i don't know how to get the zones out of order in logic tests
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/62490#issuecomment-805276172>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEMXVOR4R75Q2UCHNKFLSALTFEB7HANCNFSM4ZV7BQYQ>
> .
>

otan committed:
delegate: order AZs in SHOW REGIONS commands

Release note (sql change): Availability zones were not ordered when
using the SHOW REGIONS set of commands. This has now been changed to be
ordered.

otan commented:
bors r=ajstorm
